crates/bindings-csharp/BSATN.Runtime/BSATN/Int256.cs

```diff
@@ -0,0 +1,134 @@
+            }
+            else
+            {
+                throw new ArgumentException();
+            }
+        }
+
+        /// <inheritdoc cref="IComparable{T}.CompareTo(T)" />
+        public int CompareTo(Int256 value)
+        {
+            if (this < value)
+            {
+                return -1;
+            }
+            else if (this > value)
+            {
+                return 1;
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThan(TSelf, TOther)" />
+        public static bool operator <(Int256 left, Int256 right)
+        {
+            if (IsNegative(left) == IsNegative(right))
+            {
+                return (left._upper < right._upper)
+                    || ((left._upper == right._upper) && (left._lower < right._lower));
+            }
+            else
+            {
+                return IsNegative(left);
+            }
+        }
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_GreaterThan(TSelf, TOther)" />
+        public static bool operator >(Int256 left, Int256 right)
+        {
+            if (IsNegative(left) == IsNegative(right))
+            {
+                return (left._upper > right._upper)
+                    || ((left._upper == right._upper) && (left._lower > right._lower));
+            }
+            else
+            {
+                return IsNegative(right);
+            }
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.IsNegative(TSelf)" />
+
+        public static bool IsNegative(Int256 value) =>
+#if NET7_0_OR_GREATER
+            ((long)(value._upper >> 64))
+#else
+            (long)value._upper.Upper
+#endif
+            < 0;
+
+        //
+        // IEqualityOperators
+        //
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Equality(TSelf, TOther)" />
+        public static bool operator ==(Int256 left, Int256 right) => (left._lower == right._lower) && (left._upper == right._upper);
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Inequality(TSelf, TOther)" />
+        public static bool operator !=(Int256 left, Int256 right) => (left._lower != right._lower) || (left._upper != right._upper);
+
+        /// <inheritdoc cref="object.Equals(object?)" />
+        public override bool Equals([NotNullWhen(true)] object? obj)
+        {
+            return (obj is Int256 other) && Equals(other);
+        }
+
+        /// <inheritdoc cref="IEquatable{T}.Equals(T)" />
+        public bool Equals(Int256 x) => _upper == x._upper && _lower == x._lower;
+
+        /// <inheritdoc cref="object.GetHashCode()" />
+        public override int GetHashCode() => HashCode.Combine(_lower, _upper);
+
+        /// <inheritdoc cref="object.ToString()" />
+        public override string ToString() => $"Int256({_upper},{_lower})";
```

Comment:
Do we want to provide proper ToString() maybe? Should be easy to do via BigInteger - not the most efficient method perhaps, but should be fine for stringification.